### ISSUE TITLE
Loser Bracket bug

### ISSUE THREAD

**Issue by efkay1** [Drarig29/brackets-viewer.js#85]
Hi, I just played around with the brackets-viewer demo and found the following:

![grafik](https://user-images.githubusercontent.com/11843437/160628891-a8bb1066-25a5-4009-8b49-6b566c5eec3e.png)

'LB 1.2: Loser of WB 1.2' is not set even though there is a loser set in WB 1.2 (named '4')
Instead, Loser of WB 1.2 goes to the second slot of LB 1.1

I also found that this bug extends with bigger brackets:
![grafik](https://user-images.githubusercontent.com/11843437/160630039-21ed8db5-cda9-4282-b439-4037c1f26c78.png)

Unfortunately I am not fit enough yet with this to fix it by myself. I also could imagine that this is only an issue of the demo that generates the brackets as the data is usually coming from brackets-manager / brackets-model.
Also I have to note I tested the demo with flask/jinja engine as an extended html template.

Best regards

**Comment by Drarig29** [comment]
Hi! Thanks for testing the project!

You're right, this kind of bug usually comes from the `InMemoryDatabase` which is used only in the browser... All the unit tests are done with another type of storage so I tend not to find this kind of bug.

I just fixed a bug really close to this yesterday: https://github.com/Drarig29/brackets-manager.js/discussions/125

I'll have a look :wink:

**Comment by Drarig29** [comment]
This one is tough... I can only reproduce it every other time...

I fixed something which looks the same yesterday, but this time it's not easily reproducible.

**Comment by Drarig29** [comment]
Please be sure you are using the latest version of the viewer, I think I fixed it with the latest release: https://github.com/Drarig29/brackets-viewer.js/releases/tag/v1.4.5

**Comment by Drarig29** [comment]
No, actually I tried it again and it still does this. I'll need to investigate more.
